dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Controllers/PatientController.cs

```diff
@@ -109,18 +156,21 @@ private static IList<TimelineEntryViewModel> ConvertToTimelineEntryViewModels(Ti
                         if (schedule.LocalDateTime >= snapshot.EffectiveStartInstant.InUtc().LocalDateTime &&
                             schedule.LocalDateTime <= snapshot.EffectiveEndInstant.InUtc().LocalDateTime)
                         {
-                            // Add a timeline entry for the schedule
+                            // Add a timeline entry for the schedule. 
+                            // Count all occurences for a frequency task, and only in-window one for a scheduled task
                             timelineEntries.Add(new TimelineEntryViewModel
                             {
                                 TaskId = task.TaskId,
                                 TaskName = task.TaskName,
                                 TaskImageUrl = task.TaskImageUrl,
                                 LocalDateTime = schedule.LocalDateTime,
-                                ScheduleType = schedule.Type
+                                ScheduleType = schedule.Type,
+                                CompletionMetrics = snapshot.CompletionMetrics,
+                                OccurrenceCount = schedule.Occurrences?.Count(o => o.InWindow || snapshot.CompletionMetrics.CompletionType == ActionPlanTaskCompletionType.Frequency) ?? 0
```

Comment:
It doesn't count the extra occurrence for a "completion", does it? or the completions returned in a separate array?
Also I think you talked about this briefly, for a frequency based task it would only count the occurrences/day?